### PR TITLE
Update gcc to 10.

### PR DESCRIPTION
**Pull request by qihqi**
Also use unversioned clang-format to be more flexible in updating distro version.

### CODE REVIEW

File: infra/ansible/e2e_tests.Dockerfile [hunk 1]
```diff
@@ -1,8 +1,12 @@
 ARG python_version=3.8
-ARG debian_version=buster
+ARG debian_version=bullseye
 
 FROM python:${python_version}-${debian_version} AS build
 
+# install gcc10
+RUN apt-get update
+RUN apt-get install -y gcc-10
```
Comment (will-cromar) [inline infra/ansible/e2e_tests.Dockerfile:8]: Should gcc-10 be installed by ansible instead? This image installs all of the packages from the `build` stage.

File: infra/ansible/config/apt.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ apt:
       - rename
       - vim
       - wget
-      - clang-format-7
+      - clang-format
       - lcov
```
Comment (JackCaoG) [inline infra/ansible/config/apt.yaml:17]: can we add gcc-10 here directly?

Comment (will-cromar) [reply infra/ansible/config/apt.yaml:17]: gcc-10 is already included in the build stage for `aarch64`. Maybe just move it to `build_common`?

Comment (qihqi) [reply infra/ansible/config/apt.yaml:19]: done.

**Review by JackCaoG — APPROVED** [review]

**Pull request merged**